Add a note to a few legacy-ish list interfaces

Ahmed-Hakeem opened the pull request:
<!-- 🙌 Thanks for contributing to MDN Web Docs. Adding details below will help us to merge your PR faster. -->

### Description

Fixed the provided example since it doesn't contain item method, and added the difference between accessing item using this method and using regular property accessor

Ahmed-Hakeem committed:
fix CSSRuleList item() example

Ahmed-Hakeem committed:
fix CSSRuleList item() example

Ahmed-Hakeem committed:
Update index.md

Elchi3 reviewed (changes requested):
Thanks for your PR! A few suggestions.

(Much like DOMStringList, this interface is a legacy feature, see https://stackoverflow.com/questions/74630989/why-use-domstringlist-rather-than-an-array/74641156#74641156)

Ahmed-Hakeem committed:
Rewording item() example

Co-authored-by: Florian Scholz <fs@florianscholz.com>

Ahmed-Hakeem committed:
Rewording item() example

Co-authored-by: Florian Scholz <fs@florianscholz.com>

Ahmed-Hakeem committed:
Rewording item() example

Co-authored-by: Florian Scholz <fs@florianscholz.com>

Ahmed-Hakeem commented:
> Thanks for your PR! A few suggestions.
> 
> (Much like DOMStringList, this interface is a legacy feature, see https://stackoverflow.com/questions/74630989/why-use-domstringlist-rather-than-an-array/74641156#74641156)

Thank you for your suggestions❤️, how about making a PR to add this hint to the CSSRulesList page, and others .... or extending this PR with this hint

Elchi3 commented:
> Thank you for your suggestions❤️, how about making a PR to add this hint to the CSSRulesList page, and others .... or extending this PR with this hint

You're welcome. If you like you can extend this PR! I made a PR to DOMStringList recently: https://github.com/mdn/content/pull/22750

Ahmed-Hakeem commented:
> > Thank you for your suggestions❤️, how about making a PR to add this hint to the CSSRulesList page, and others .... or extending this PR with this hint
> 
> 
> 
> You're welcome. If you like you can extend this PR! I made a PR to DOMStringList recently: https://github.com/mdn/content/pull/22750

Sure

Ahmed-Hakeem committed:
add legacy note for cssRuleList

Ahmed-Hakeem committed:
Merge branch 'fix-cssRuleListItem-method' of https://github.com/Ahmed-Hakeem/content into fix-cssRuleListItem-method

Ahmed-Hakeem committed:
add legacy note to old readonly collection interfaces

Ahmed-Hakeem committed:
fix linting issues

sideshowbarker commented on files/en-us/web/api/cssrulelist/index.md:
```diff
@@ -20,6 +20,8 @@ To edit the underlying rules returned by `CSSRule` objects, use {{domxref("CSSSt
 
 The interface has no constructor. An instance of `CSSRuleList` is returned by {{domxref("CSSStyleSheet.cssRules")}} and {{domxref("CSSKeyframesRule.cssRules")}}.
 
+> **Note:** This interface is a legacy type that existed because of a [historical accident](https://stackoverflow.com/questions/74630989/why-use-domstringlist-rather-than-an-array/74641156#74641156) and only kept for not breaking the code that have already used it. Modern APIs use Types that wrap around ECMAScript array types so you can treat them like an ECMAScript array, and at the same time im ...
```
```suggestion
> **Note:** This interface is a legacy type that exists because of a [historical accident](https://stackoverflow.com/questions/74630989/why-use-domstringlist-rather-than-an-array/74641156#74641156) and is only continues to be supported in order not to break code that's already using it. Modern APIs use types that wrap around ECMAScript array types so you can treat them like ECMAScript arrays, and at the same time impose additional semantics on their usage (such as making their items read-only).
```

sideshowbarker committed:
Update files/en-us/web/api/cssrulelist/index.md

sideshowbarker commented on files/en-us/web/api/cssrulelist/item/index.md:
```diff
@@ -32,11 +32,15 @@ A {{domxref("CSSRule")}}.
 ## Examples
 
-In the following example the first item in the {{domxref("CSSRuleList")}} named `myRules` is printed to the console.
+In the following example we assume that the `myRules` list has three items only.
```
```suggestion
In the following example, we assume that the `myRules` list has three items only.
```

sideshowbarker committed:
Update files/en-us/web/api/cssrulelist/item/index.md

sideshowbarker commented on files/en-us/web/api/cssstyledeclaration/index.md:
```diff
@@ -21,6 +21,8 @@ A `CSSStyleDeclaration` object can be exposed using three different APIs:
 - Via the {{DOMxRef("CSSStyleSheet")}} API. For example, `document.styleSheets[0].cssRules[0].style` returns a `CSSStyleDeclaration` object on the first CSS rule in the document's first stylesheet.
 - Via {{DOMxRef("Window.getComputedStyle()")}}, which exposes the `CSSStyleDeclaration` object as a **read-only** interface.
 
```
```suggestion
> **Note:** This interface is a legacy type that exists because of a [historical accident](https://stackoverflow.com/questions/74630989/why-use-domstringlist-rather-than-an-array/74641156#74641156) and is only continues to be supported in order not to break code that's already using it. Modern APIs use types that wrap around ECMAScript array types so you can treat them like ECMAScript arrays, and at the same time impose additional semantics on their usage (such as making their items read-only).
```

sideshowbarker commented on files/en-us/web/api/cssstyledeclaration/index.md:
```diff
@@ -21,6 +21,8 @@ A `CSSStyleDeclaration` object can be exposed using three different APIs:
 - Via the {{DOMxRef("CSSStyleSheet")}} API. For example, `document.styleSheets[0].cssRules[0].style` returns a `CSSStyleDeclaration` object on the first CSS rule in the document's first stylesheet.
 - Via {{DOMxRef("Window.getComputedStyle()")}}, which exposes the `CSSStyleDeclaration` object as a **read-only** interface.
 
+> **Note:** This interface is a legacy type that existed because of a [historical accident](https://stackoverflow.com/questions/74630989/why-use-domstringlist-rather-than-an-array/74641156#74641156) and only kept for not breaking the code that have already used it. Modern APIs use Types that wrap around ECMAScript array types so you can treat them like an ECMAScript array, and at the same time im ...
```
```suggestion
> **Note:** This interface is a legacy type that exists because of a [historical accident](https://stackoverflow.com/questions/74630989/why-use-domstringlist-rather-than-an-array/74641156#74641156) and is only continues to be supported in order not to break code that's already using it. Modern APIs use types that wrap around ECMAScript array types so you can treat them like ECMAScript arrays, and at the same time impose additional semantics on their usage (such as making their items read-only).
```

sideshowbarker committed:
Update files/en-us/web/api/cssstyledeclaration/index.md

sideshowbarker commented on files/en-us/web/api/domtokenlist/index.md:
```diff
@@ -14,6 +14,8 @@ The **`DOMTokenList`** interface represents a set of space-separated tokens. Suc
 
 A `DOMTokenList` is indexed beginning with `0` as with JavaScript {{jsxref("Array")}} objects. `DOMTokenList` is always case-sensitive.
 
+> **Note:** This interface is a legacy type that existed because of a [historical accident](https://stackoverflow.com/questions/74630989/why-use-domstringlist-rather-than-an-array/74641156#74641156) and only kept for not breaking the code that have already used it. Modern APIs use Types that wrap around ECMAScript array types so you can treat them like an ECMAScript array, and at the same time im ...
```
```suggestion
> **Note:** This interface is a legacy type that exists because of a [historical accident](https://stackoverflow.com/questions/74630989/why-use-domstringlist-rather-than-an-array/74641156#74641156) and is only continues to be supported in order not to break code that's already using it. Modern APIs use types that wrap around ECMAScript array types so you can treat them like ECMAScript arrays, and at the same time impose additional semantics on their usage (such as making their items read-only).
```

sideshowbarker committed:
Update files/en-us/web/api/domtokenlist/index.md

sideshowbarker commented on files/en-us/web/api/filelist/index.md:
````diff
@@ -25,6 +25,8 @@ The following line of code fetches the first file in the node's file list as a [
 const file = document.getElementById('fileItem').files[0];
 ```
 
+> **Note:** This interface is a legacy type that existed because of a [historical accident](https://stackoverflow.com/questions/74630989/why-use-domstringlist-rather-than-an-array/74641156#74641156) and only kept for not breaking the code that have already used it. Modern APIs use Types that wrap around ECMAScript array types so you can treat them like an ECMAScript array, and at the same time im ...
````
```suggestion
> **Note:** This interface is a legacy type that exists because of a [historical accident](https://stackoverflow.com/questions/74630989/why-use-domstringlist-rather-than-an-array/74641156#74641156) and is only continues to be supported in order not to break code that's already using it. Modern APIs use types that wrap around ECMAScript array types so you can treat them like ECMAScript arrays, and at the same time impose additional semantics on their usage (such as making their items read-only).
```

sideshowbarker committed:
Update files/en-us/web/api/filelist/index.md

sideshowbarker commented on files/en-us/web/api/htmlcollection/index.md:
```diff
@@ -20,6 +20,8 @@ The **`HTMLCollection`** interface represents a generic collection (array-like o
 
 An `HTMLCollection` in the HTML DOM is live; it is automatically updated when the underlying document is changed. For this reason it is a good idea to make a copy (e.g., using {{jsxref("Array/from", "Array.from")}}) to iterate over if adding, moving, or removing nodes.
 
+> **Note:** This interface is a legacy type that existed because of a [historical accident](https://stackoverflow.com/questions/74630989/why-use-domstringlist-rather-than-an-array/74641156#74641156) and only kept for not breaking the code that have already used it. Modern APIs use Types that wrap around ECMAScript array types so you can treat them like an ECMAScript array, and at the same time im ...
```
```suggestion
> **Note:** This interface is a legacy type that exists because of a [historical accident](https://stackoverflow.com/questions/74630989/why-use-domstringlist-rather-than-an-array/74641156#74641156) and is only continues to be supported in order not to break code that's already using it. Modern APIs use types that wrap around ECMAScript array types so you can treat them like ECMAScript arrays, and at the same time impose additional semantics on their usage (such as making their items read-only).
```

sideshowbarker commented on files/en-us/web/api/htmlselectelement/index.md:
```diff
@@ -17,6 +17,8 @@ The **`HTMLSelectElement`** interface represents a {{HTMLElement("select")}} HTM
 
 {{InheritanceDiagram}}
 
+> **Note:** This interface is a legacy type that existed because of a [historical accident](https://stackoverflow.com/questions/74630989/why-use-domstringlist-rather-than-an-array/74641156#74641156) and only kept for not breaking the code that have already used it. Modern APIs use Types that wrap around ECMAScript array types so you can treat them like an ECMAScript array, and at the same time im ...
```
```suggestion
> **Note:** This interface is a legacy type that exists because of a [historical accident](https://stackoverflow.com/questions/74630989/why-use-domstringlist-rather-than-an-array/74641156#74641156) and is only continues to be supported in order not to break code that's already using it. Modern APIs use types that wrap around ECMAScript array types so you can treat them like ECMAScript arrays, and at the same time impose additional semantics on their usage (such as making their items read-only).
```

sideshowbarker commented on files/en-us/web/api/medialist/index.md:
```diff
@@ -17,6 +17,8 @@ The **`MediaList`** interface represents the media queries of a stylesheet, e.g.
 
 > **Note:** `MediaList` is a live list; updating the list using properties or methods listed below will immediately update the behavior of the document.
 
+> **Note:** This interface is a legacy type that existed because of a [historical accident](https://stackoverflow.com/questions/74630989/why-use-domstringlist-rather-than-an-array/74641156#74641156) and only kept for not breaking the code that have already used it. Modern APIs use Types that wrap around ECMAScript array types so you can treat them like an ECMAScript array, and at the same time im ...
```
```suggestion
> **Note:** This interface is a legacy type that exists because of a [historical accident](https://stackoverflow.com/questions/74630989/why-use-domstringlist-rather-than-an-array/74641156#74641156) and is only continues to be supported in order not to break code that's already using it. Modern APIs use types that wrap around ECMAScript array types so you can treat them like ECMAScript arrays, and at the same time impose additional semantics on their usage (such as making their items read-only).
```

sideshowbarker commented on files/en-us/web/api/mimetypearray/index.md:
```diff
@@ -15,6 +15,8 @@ browser-compat: api.MimeTypeArray
 
 The **`MimeTypeArray`** interface returns an array of {{domxref('MimeType')}} instances, each of which contains information about a supported browser plugins. This object is returned by {{domxref("Navigator.mimeTypes")}}.
 
+> **Note:** This interface is a legacy type that existed because of a [historical accident](https://stackoverflow.com/questions/74630989/why-use-domstringlist-rather-than-an-array/74641156#74641156) and only kept for not breaking the code that have already used it. Modern APIs use Types that wrap around ECMAScript array types so you can treat them like an ECMAScript array, and at the same time im ...
```
```suggestion
> **Note:** This interface is a legacy type that exists because of a [historical accident](https://stackoverflow.com/questions/74630989/why-use-domstringlist-rather-than-an-array/74641156#74641156) and is only continues to be supported in order not to break code that's already using it. Modern APIs use types that wrap around ECMAScript array types so you can treat them like ECMAScript arrays, and at the same time impose additional semantics on their usage (such as making their items read-only).
```

sideshowbarker commented on files/en-us/web/api/namednodemap/index.md:
```diff
@@ -16,6 +16,8 @@ A `NamedNodeMap` object is _live_ and will thus be auto-updated if changes are m
 
 > **Note:** Although called `NamedNodeMap`, this interface doesn't deal with {{domxref("Node")}} objects but with {{domxref("Attr")}} objects, which are a specialized class of {{domxref("Node")}} objects.
 
+> **Note:** This interface is a legacy type that existed because of a [historical accident](https://stackoverflow.com/questions/74630989/why-use-domstringlist-rather-than-an-array/74641156#74641156) and only kept for not breaking the code that have already used it. Modern APIs use Types that wrap around ECMAScript array types so you can treat them like an ECMAScript array, and at the same time im ...
```
```suggestion
> **Note:** This interface is a legacy type that exists because of a [historical accident](https://stackoverflow.com/questions/74630989/why-use-domstringlist-rather-than-an-array/74641156#74641156) and is only continues to be supported in order not to break code that's already using it. Modern APIs use types that wrap around ECMAScript array types so you can treat them like ECMAScript arrays, and at the same time impose additional semantics on their usage (such as making their items read-only).
```

sideshowbarker commented on files/en-us/web/api/nodelist/index.md:
```diff
@@ -16,6 +16,8 @@ browser-compat: api.NodeList
 
 > **Note:** Although `NodeList` is not an `Array`, it is possible to iterate over it with `forEach()`. It can also be converted to a real `Array` using {{jsxref("Array.from()")}}.
 
+> **Note:** This interface is a legacy type that existed because of a [historical accident](https://stackoverflow.com/questions/74630989/why-use-domstringlist-rather-than-an-array/74641156#74641156) and only kept for not breaking the code that have already used it. Modern APIs use Types that wrap around ECMAScript array types so you can treat them like an ECMAScript array, and at the same time im ...
```
```suggestion
> **Note:** This interface is a legacy type that exists because of a [historical accident](https://stackoverflow.com/questions/74630989/why-use-domstringlist-rather-than-an-array/74641156#74641156) and is only continues to be supported in order not to break code that's already using it. Modern APIs use types that wrap around ECMAScript array types so you can treat them like ECMAScript arrays, and at the same time impose additional semantics on their usage (such as making their items read-only).
```

sideshowbarker commented on files/en-us/web/api/plugin/index.md:
```diff
@@ -19,6 +19,8 @@ The `Plugin` interface provides information about a browser plugin.
 
 > **Note:** Own properties of `Plugin` objects are no longer enumerable in the latest browser versions.
 
+> **Note:** This interface is a legacy type that existed because of a [historical accident](https://stackoverflow.com/questions/74630989/why-use-domstringlist-rather-than-an-array/74641156#74641156) and only kept for not breaking the code that have already used it. Modern APIs use Types that wrap around ECMAScript array types so you can treat them like an ECMAScript array, and at the same time im ...
```
```suggestion
> **Note:** This interface is a legacy type that exists because of a [historical accident](https://stackoverflow.com/questions/74630989/why-use-domstringlist-rather-than-an-array/74641156#74641156) and is only continues to be supported in order not to break code that's already using it. Modern APIs use types that wrap around ECMAScript array types so you can treat them like ECMAScript arrays, and at the same time impose additional semantics on their usage (such as making their items read-only).
```

sideshowbarker commented on files/en-us/web/api/pluginarray/index.md:
```diff
@@ -18,6 +18,8 @@ The `PluginArray` interface is used to store a list of {{DOMxRef("Plugin")}} obj
 
 > **Note:** Own properties of `PluginArray` objects are no longer enumerable in the latest browser versions.
 
+> **Note:** This interface is a legacy type that existed because of a [historical accident](https://stackoverflow.com/questions/74630989/why-use-domstringlist-rather-than-an-array/74641156#74641156) and only kept for not breaking the code that have already used it. Modern APIs use Types that wrap around ECMAScript array types so you can treat them like an ECMAScript array, and at the same time im ...
```
```suggestion
> **Note:** This interface is a legacy type that exists because of a [historical accident](https://stackoverflow.com/questions/74630989/why-use-domstringlist-rather-than-an-array/74641156#74641156) and is only continues to be supported in order not to break code that's already using it. Modern APIs use types that wrap around ECMAScript array types so you can treat them like ECMAScript arrays, and at the same time impose additional semantics on their usage (such as making their items read-only).
```

sideshowbarker commented on files/en-us/web/api/speechgrammarlist/index.md:
```diff
@@ -20,6 +20,8 @@ The **`SpeechGrammarList`** interface of the [Web Speech API](/en-US/docs/Web/AP
 
 Grammar is defined using [JSpeech Grammar Format](https://www.w3.org/TR/jsgf/) (**JSGF**.) Other formats may also be supported in the future.
 
+> **Note:** This interface is a legacy type that existed because of a [historical accident](https://stackoverflow.com/questions/74630989/why-use-domstringlist-rather-than-an-array/74641156#74641156) and only kept for not breaking the code that have already used it. Modern APIs use Types that wrap around ECMAScript array types so you can treat them like an ECMAScript array, and at the same time im ...
```
```suggestion
> **Note:** This interface is a legacy type that exists because of a [historical accident](https://stackoverflow.com/questions/74630989/why-use-domstringlist-rather-than-an-array/74641156#74641156) and is only continues to be supported in order not to break code that's already using it. Modern APIs use types that wrap around ECMAScript array types so you can treat them like ECMAScript arrays, and at the same time impose additional semantics on their usage (such as making their items read-only).
```

sideshowbarker commented on files/en-us/web/api/speechrecognitionresult/index.md:
```diff
@@ -17,6 +17,8 @@ browser-compat: api.SpeechRecognitionResult
 
 The **`SpeechRecognitionResult`** interface of the [Web Speech API](/en-US/docs/Web/API/Web_Speech_API) represents a single recognition match, which may contain multiple {{domxref("SpeechRecognitionAlternative")}} objects.
 
+> **Note:** This interface is a legacy type that existed because of a [historical accident](https://stackoverflow.com/questions/74630989/why-use-domstringlist-rather-than-an-array/74641156#74641156) and only kept for not breaking the code that have already used it. Modern APIs use Types that wrap around ECMAScript array types so you can treat them like an ECMAScript array, and at the same time im ...
```
```suggestion
> **Note:** This interface is a legacy type that exists because of a [historical accident](https://stackoverflow.com/questions/74630989/why-use-domstringlist-rather-than-an-array/74641156#74641156) and is only continues to be supported in order not to break code that's already using it. Modern APIs use types that wrap around ECMAScript array types so you can treat them like ECMAScript arrays, and at the same time impose additional semantics on their usage (such as making their items read-only).
```

sideshowbarker commented on files/en-us/web/api/speechrecognitionresultlist/index.md:
```diff
@@ -17,6 +17,8 @@ browser-compat: api.SpeechRecognitionResultList
 
 The **`SpeechRecognitionResultList`** interface of the [Web Speech API](/en-US/docs/Web/API/Web_Speech_API) represents a list of {{domxref("SpeechRecognitionResult")}} objects, or a single one if results are being captured in {{domxref("SpeechRecognition.continuous","continuous")}} mode.
 
+> **Note:** This interface is a legacy type that existed because of a [historical accident](https://stackoverflow.com/questions/74630989/why-use-domstringlist-rather-than-an-array/74641156#74641156) and only kept for not breaking the code that have already used it. Modern APIs use Types that wrap around ECMAScript array types so you can treat them like an ECMAScript array, and at the same time im ...
```
```suggestion
> **Note:** This interface is a legacy type that exists because of a [historical accident](https://stackoverflow.com/questions/74630989/why-use-domstringlist-rather-than-an-array/74641156#74641156) and is only continues to be supported in order not to break code that's already using it. Modern APIs use types that wrap around ECMAScript array types so you can treat them like ECMAScript arrays, and at the same time impose additional semantics on their usage (such as making their items read-only).
```

sideshowbarker commented on files/en-us/web/api/stylesheetlist/index.md:
```diff
@@ -15,6 +15,8 @@ The `StyleSheetList` interface represents a list of {{domxref("CSSStyleSheet")}}
 
 It is an array-like object but can't be iterated over using {{jsxref("Array")}} methods. However it can be iterated over in a standard {{jsxref("Statements/for", "for")}} loop over its indices, or converted to an {{jsxref("Array")}}.
 
+> **Note:** This interface is a legacy type that existed because of a [historical accident](https://stackoverflow.com/questions/74630989/why-use-domstringlist-rather-than-an-array/74641156#74641156) and only kept for not breaking the code that have already used it. Modern APIs use Types that wrap around ECMAScript array types so you can treat them like an ECMAScript array, and at the same time im ...
```
```suggestion
> **Note:** This interface is a legacy type that exists because of a [historical accident](https://stackoverflow.com/questions/74630989/why-use-domstringlist-rather-than-an-array/74641156#74641156) and is only continues to be supported in order not to break code that's already using it. Modern APIs use types that wrap around ECMAScript array types so you can treat them like ECMAScript arrays, and at the same time impose additional semantics on their usage (such as making their items read-only).
```

sideshowbarker commented on files/en-us/web/api/touchlist/index.md:
```diff
@@ -18,6 +18,8 @@ browser-compat: api.TouchList
 
 The **`TouchList`** interface represents a list of contact points on a touch surface. For example, if the user has three fingers on the touch surface (such as a screen or trackpad), the corresponding `TouchList` object would have one {{domxref("Touch")}} object for each finger, for a total of three entries.
 
+> **Note:** This interface is a legacy type that existed because of a [historical accident](https://stackoverflow.com/questions/74630989/why-use-domstringlist-rather-than-an-array/74641156#74641156) and only kept for not breaking the code that have already used it. Modern APIs use Types that wrap around ECMAScript array types so you can treat them like an ECMAScript array, and at the same time im ...
```
```suggestion
> **Note:** This interface is a legacy type that exists because of a [historical accident](https://stackoverflow.com/questions/74630989/why-use-domstringlist-rather-than-an-array/74641156#74641156) and is only continues to be supported in order not to break code that's already using it. Modern APIs use types that wrap around ECMAScript array types so you can treat them like ECMAScript arrays, and at the same time impose additional semantics on their usage (such as making their items read-only).
```

sideshowbarker committed:
Apply suggestions from code review

Elchi3 reviewed (changes requested):
I think it is wrong to call these interfaces legacy:
- CSSStyleDeclaration
- DOMTokenList
- HTMLSelectElement
- MediaList
- NodeList
- SpeechGrammarList
- SpeechRecognitionResultList
- SpeechRecognitionResult

I'm also not sure if it should be called "historical accident". That seems to add unnecessary judgement.

Ahmed-Hakeem commented:
> I think it is wrong to call these interfaces legacy:
> 
> * CSSStyleDeclaration
> * DOMTokenList
> * HTMLSelectElement
> * MediaList
> * NodeList
> * SpeechGrammarList
> * SpeechRecognitionResultList
> * SpeechRecognitionResult
> 
> I'm also not sure if it should be called "historical accident". That seems to add unnecessary judgement.

How about 
"This interface was a [try to achieve unmodifiable list](https://stackoverflow.com/questions/74630989/why-use-domstringlist-rather-than-an-array/74641156#74641156) and is only continues to be supported in order not to break code that's already using it. Modern APIs use types that wrap around ECMAScript array types instead, so you can treat them like ECMAScript arrays, and at the same time impose additional semantics on their usage (such as making their items read-only)."

Elchi3 commented:
That sounds better. Slightly corrected version:

"This interface was an [attempt to create an unmodifiable list](https://stackoverflow.com/questions/74630989/why-use-domstringlist-rather-than-an-array/74641156#74641156) and only continues to be supported to not break code that's already using it. Modern APIs use types that wrap around ECMAScript array types instead, so you can treat them like ECMAScript arrays, and at the same time impose additional semantics on their usage (such as making their items read-only)."

Ahmed-Hakeem committed:
rephrase the history of read-only collection interfaces note

Elchi3 commented:
Thanks for the rewording. I'm still not quite sure if this note should be added to all interfaces in this PR.
I still have doubts about these:

> * CSSStyleDeclaration
> * DOMTokenList
> * HTMLSelectElement
> * MediaList
> * NodeList
> * SpeechGrammarList
> * SpeechRecognitionResultList
> * SpeechRecognitionResult

Maybe @sideshowbarker has thoughts.

sideshowbarker commented:
I'll take a look again

sideshowbarker commented:
> Thanks for the rewording. I'm still not quite sure if this note should be added to all interfaces in this PR. I still have doubts about these:
> 
> > * CSSStyleDeclaration
> > * DOMTokenList
> > * HTMLSelectElement
> > * MediaList
> > * NodeList
> > * SpeechGrammarList
> > * SpeechRecognitionResultList
> > * SpeechRecognitionResult
> 
> Maybe @sideshowbarker has thoughts.

I haven’t managed to make time to look at the relevant specs yet, but regardless I think it anyway be good to have @domenic take a look at this too

domenic commented on files/en-us/web/api/cssstyledeclaration/index.md:
```diff
@@ -21,6 +21,8 @@ A `CSSStyleDeclaration` object can be exposed using three different APIs:
 - Via the {{DOMxRef("CSSStyleSheet")}} API. For example, `document.styleSheets[0].cssRules[0].style` returns a `CSSStyleDeclaration` object on the first CSS rule in the document's first stylesheet.
 - Via {{DOMxRef("Window.getComputedStyle()")}}, which exposes the `CSSStyleDeclaration` object as a **read-only** interface.
 
+> **Note:** This interface was an [attempt to create an unmodifiable list](https://stackoverflow.com/questions/74630989/why-use-domstringlist-rather-than-an-array/74641156#74641156) and only continues to be supported to not break code that's already using it. Modern APIs use types that wrap around ECMAScript array types instead, so you can treat them like ECMAScript arrays, and at the same time i ...
```
I think this class does too much other stuff to add this warning for. If we were defining it today, we'd separate out the other functionality from the list stuff, but since they're mixed together, I think it'd be a mistake to speak about this as legacy-ish.

domenic commented on files/en-us/web/api/domtokenlist/index.md:
```diff
@@ -14,6 +14,8 @@ The **`DOMTokenList`** interface represents a set of space-separated tokens. Suc
 
 A `DOMTokenList` is indexed beginning with `0` as with JavaScript {{jsxref("Array")}} objects. `DOMTokenList` is always case-sensitive.
 
+> **Note:** This interface was an [attempt to create an unmodifiable list](https://stackoverflow.com/questions/74630989/why-use-domstringlist-rather-than-an-array/74641156#74641156) and only continues to be supported to not break code that's already using it. Modern APIs use types that wrap around ECMAScript array types instead, so you can treat them like ECMAScript arrays, and at the same time i ...
```
Similarly I think this does too many additional non-Array-like things to get this note.

domenic commented on files/en-us/web/api/htmlselectelement/index.md:
```diff
@@ -17,6 +17,8 @@ The **`HTMLSelectElement`** interface represents a {{HTMLElement("select")}} HTM
 
 {{InheritanceDiagram}}
 
+> **Note:** This interface was an [attempt to create an unmodifiable list](https://stackoverflow.com/questions/74630989/why-use-domstringlist-rather-than-an-array/74641156#74641156) and only continues to be supported to not break code that's already using it. Modern APIs use types that wrap around ECMAScript array types instead, so you can treat them like ECMAScript arrays, and at the same time i ...
```
The fact that this has indexed access is almost incidental; I think this note is strange here.

domenic commented on files/en-us/web/api/medialist/index.md:
```diff
@@ -17,6 +17,8 @@ The **`MediaList`** interface represents the media queries of a stylesheet, e.g.
 
 > **Note:** `MediaList` is a live list; updating the list using properties or methods listed below will immediately update the behavior of the document.
 
+> **Note:** This interface was an [attempt to create an unmodifiable list](https://stackoverflow.com/questions/74630989/why-use-domstringlist-rather-than-an-array/74641156#74641156) and only continues to be supported to not break code that's already using it. Modern APIs use types that wrap around ECMAScript array types instead, so you can treat them like ECMAScript arrays, and at the same time i ...
```
It's weird to say that this is an attempt to create an unmodifiable list, when in fact it's very modifiable (via appendMedium()/deleteMedium()). It's definitely legacy though; it's just like an observable array but worse.

domenic commented on files/en-us/web/api/namednodemap/index.md:
```diff
@@ -16,6 +16,8 @@ A `NamedNodeMap` object is _live_ and will thus be auto-updated if changes are m
 
 > **Note:** Although called `NamedNodeMap`, this interface doesn't deal with {{domxref("Node")}} objects but with {{domxref("Attr")}} objects, which are a specialized class of {{domxref("Node")}} objects.
 
+> **Note:** This interface was an [attempt to create an unmodifiable list](https://stackoverflow.com/questions/74630989/why-use-domstringlist-rather-than-an-array/74641156#74641156) and only continues to be supported to not break code that's already using it. Modern APIs use types that wrap around ECMAScript array types instead, so you can treat them like ECMAScript arrays, and at the same time i ...
```
Probably this one does too much to get this note.

domenic commented on files/en-us/web/api/plugin/index.md:
```diff
@@ -19,6 +19,8 @@ The `Plugin` interface provides information about a browser plugin.
 
 > **Note:** Own properties of `Plugin` objects are no longer enumerable in the latest browser versions.
 
+> **Note:** This interface was an [attempt to create an unmodifiable list](https://stackoverflow.com/questions/74630989/why-use-domstringlist-rather-than-an-array/74641156#74641156) and only continues to be supported to not break code that's already using it. Modern APIs use types that wrap around ECMAScript array types instead, so you can treat them like ECMAScript arrays, and at the same time i ...
```
Does too much other stuff.

domenic commented on files/en-us/web/api/speechgrammarlist/index.md:
```diff
@@ -20,6 +20,8 @@ The **`SpeechGrammarList`** interface of the [Web Speech API](/en-US/docs/Web/AP
 
 Grammar is defined using [JSpeech Grammar Format](https://www.w3.org/TR/jsgf/) (**JSGF**.) Other formats may also be supported in the future.
 
+> **Note:** This interface was an [attempt to create an unmodifiable list](https://stackoverflow.com/questions/74630989/why-use-domstringlist-rather-than-an-array/74641156#74641156) and only continues to be supported to not break code that's already using it. Modern APIs use types that wrap around ECMAScript array types instead, so you can treat them like ECMAScript arrays, and at the same time i ...
```
Seems borderline; unsure.

Ahmed-Hakeem commented:
Thanks much, @domenic  for taking time reviewing this, we can modify the note to mention that this not the only reason behind creating these interfaces, but Let's stick with removing it from these pages, and aggregating all your suggestions @Elchi3 @sideshowbarker @domenic  we will remove this note from : 

- CSSStyleDeclaration
- DOMTokenList
- HTMLSelectElement
- MediaList
- NodeList
- SpeechGrammarList
- SpeechRecognitionResult
- NamedNodeMap
- Plugin


Does this note fit SpeechRecognitionResultList?.... what do you think @Elchi3 ?

Ahmed-Hakeem commented:
If I got your points well we should add PluginArray to this list as well

Ahmed-Hakeem committed:
remove read-only collections history note

remove the note about the history of creating read-only interfaces from interfaces that was designed to do more work than just achieving unmodifiable list

Ahmed-Hakeem committed:
fix linting issue

sideshowbarker commented:
Thanks @domenic and thanks @Ahmed-Hakeem for your work and perseverance on this one.

@Elchi3 I believe this is ready to go now

Ahmed-Hakeem commented:
> Thanks @domenic and thanks @Ahmed-Hakeem for your work and perseverance on this one.
> 
> 
> 
> @Elchi3 I believe this is ready to go now

It's my pleasure ❤️

Elchi3 reviewed (approved):
Thanks everyone!
